### ISSUE TITLE
Roles are not correcly restricting.

### ISSUE THREAD

**Issue by GDuroS** [anvil-works/material-3-theme#231]
**Describe the bug**
When you create a new role and restrict to, say, "Button", that new role does not appear in the list of roles of that component. It only appears if you don't restrict.

**To Reproduce**
Steps to reproduce the behaviour:
1. Go to 'Theme > Roles'
2. Create a new Role.
3. Click on "Restrict".
4. Type any component name. Ex. 'Button'.
5. Try to select the role just created in design in the component you restricted to.

**Screenshots**
![image](https://github.com/user-attachments/assets/e1d6e0d1-0318-453e-9dfa-16178751efb0)
_Role created and restricted_
![image](https://github.com/user-attachments/assets/6d157959-7c06-455e-a066-1cfd4dddfe5d)
_Role doesn't appear in the selection box (the role is selected because it was first allowed to all components, selected, then restricted)_

**Comment by bcm628** [comment]
This is not a bug but a usability issue. This happens because "Card" is not technically the name for the M3 Card component. Restricting the roles will work if you use "m3._Components.Card".

However, this is very ugly, so we will need to think about how to make this more user friendly. Ideally, you would be able to add a role using "m3.Card" (or similar). You can't just use the name of the component because there are M3 components that have the same name as Anvil components. 

When this is changed, we'll need to update the documentation to make this clear.

**Comment by GDuroS** [comment]
Yeah, I got that this was the issue. I tried a few different approaches to see which one worked, but none did. I opened the issue as a bug mostly because I thought maybe you didn't realize about this and just to notify about it.

**Comment by bcm628** [comment]
I'm going to close this issue because the real problem is that Anvil lets you type whatever you want in that text box. There's an existing issue in the Anvil repo to restrict your choices to actual real components, that way you don't have to know the correct name of the component. I've bumped that issue as a higher priority.
